### PR TITLE
[Fix: 146] Added `api_url` field

### PR DESCRIPTION
**Pull request by uzaxirr**
Fixes: #146 

The base URL for CIVO APIs can be configured in terraform files itself.

### CODE REVIEW

**Comment by alejandrojnm** [comment]
Hi @uzaxirr thanks for the PR, I will download and test the change in our local testing platform.

**Comment by uzaxirr** [comment]
> Hi @uzaxirr thanks for the PR, I will download and test the change in our local testing platform.

Sure, do lemme know how it goes!

**Comment by alejandrojnm** [comment]
Can you merge master with branch @uzaxirr ?

File: civo/provider.go [hunk 1]
```diff
@@ -2,13 +2,13 @@ package civo
...
 	"fmt"
-	"log"
-	"os"
-
 	"github.com/civo/civogo"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
 )
 
+const ProdAPI = "https://api.civo.com"
```
Comment (alejandrojnm) [inline civo/provider.go:10]: can you pass this to inside `var` secction

**Comment by uzaxirr** [comment]
> Can you merge master with branch @uzaxirr ?

Sure @alejandrojnm , Resolved conflicts as well

File: civo/provider.go [hunk 2]
```diff
@@ -30,6 +30,12 @@ func Provider() *schema.Provider {
 				DefaultFunc: schema.EnvDefaultFunc("CIVO_REGION", ""),
 				Description: "If region is not set, then no region will be used and them you need expensify in every resource even if you expensify here you can overwrite in a resource.",
 			},
+			"api_endpoint": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("CIVO_API_URL", "https://api.civo.com"),
```
Comment (alejandrojnm) [inline civo/provider.go:36]: replace `"https://api.civo.com"` with `ProdAPI`

File: civo/provider.go [hunk 3]
```diff
@@ -92,19 +100,16 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
...
+		apiURL = api_url.(string)
+	} else {
+		apiURL = ProdAPI
 	}
-
+	client, err = civogo.NewClientWithURL(tokenValue, apiURL, regionValue)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[DEBUG] Civo API URL: %s\n", apiURL)
```
Comment (alejandrojnm) [inline civo/provider.go:112]: Remove this line, we have the same line at the end of the file

**Comment by alejandrojnm** [comment]
I left some comments in the code

**Comment by uzaxirr** [comment]
> I left some comments in the code

@alejandrojnm I'm Not able to see any comments, you might have not clicked the submit review button perhaps

File: civo/provider.go (hunk 1 above)
Comment (uzaxirr) [reply civo/provider.go]: Done

File: civo/provider.go (hunk 2 above)
Comment (uzaxirr) [reply civo/provider.go]: Done

File: civo/provider.go (hunk 3 above)
Comment (uzaxirr) [reply civo/provider.go]: Done

**Comment by uzaxirr** [comment]
Hey @alejandrojnm can you test it again in your local testing platform. If no more changes are required

**Comment by zulh-civo** [comment]
@alejandrojnm can we merge this?

**Comment by alejandrojnm** [comment]
I will test today, if everything is ok I will merge

**Comment by zulh-civo** [comment]
@alejandrojnm any update?

**Comment by zulh-civo** [comment]
@alejandrojnm any news?

**Review by alejandrojnm — APPROVED** [review]

**Comment by alejandrojnm** [comment]
Sorry @uzaxirr need to fix this:
```bash
civo/provider.go:104:5: don't use underscores in Go names; var api_url should be apiURL
```

File: civo/provider.go [hunk 4]
```diff
@@ -92,30 +101,22 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
...
-
-	apiURL, envExists := os.LookupEnv("CIVO_API_URL")
-	if envExists && apiURL != "" {
-		client, err = civogo.NewClientWithURL(tokenValue, apiURL, regionValue)
-		if err != nil {
-			return nil, err
-		}
-		log.Printf("[DEBUG] Civo API URL: %s\n", apiURL)
-		return client, nil
+	if api_url, ok := d.GetOk("api_url"); ok {
```
Comment (alejandrojnm) [inline civo/provider.go:104]: Here is `api_endpoint` not `api_url`

Comment (uzaxirr) [reply civo/provider.go]: Done changed the variable name

**Review by alejandrojnm — APPROVED** [review]
LGTM

**Comment by uzaxirr** [comment]
@alejandrojnm  Can we merge it now?

**Comment by alejandrojnm** [comment]
I will check tomorrow to made the release

**Comment by zulh-civo** [comment]
Hoping to see a new release for this one soon :)

**Comment by zulh-civo** [comment]
@alejandrojnm / @alessandroargentieri can we cut a release for this?

**Comment by zulh-civo** [comment]
Saw the [new release](https://github.com/civo/terraform-provider-civo/releases/tag/v1.0.24), thanks!

**Pull request merged**